### PR TITLE
[Minor] ExtractImagePatches grad fails when batch size is unknown

### PR DESCRIPTION
**Pull request by liuyipei**
Mentioned in
#5501
#3672
Solution was by @mohamedadaly

### CODE REVIEW

**Comment by liuyipei** [comment]
I signed it!

**Comment by liuyipei** [comment]
I signed it!

File: tensorflow/python/ops/array_grad.py [hunk 1]
```diff
@@ -504,6 +504,8 @@ def _ExtractImagePatchesGrad(op, grad):
   batch_size, rows_in, cols_in, channels = [
     dim.value for dim in op.inputs[0].get_shape()
   ]
+  batch_size = array_ops.shape(op.inputs[0])[0]
+
```
Comment (benoitsteiner) [inline tensorflow/python/ops/array_grad.py]: To be consistent, you should also use array_ops.shape() to extract rows_in, cols_in, and channels

**Review by benoitsteiner — CHANGES_REQUESTED** [review]

**Comment by liuyipei** [comment]
@benoitsteiner 
I changed `channels` as per your request;  sadly, there is a distinction for the others. 
`rows_out`, `rows_in`, and etc are actually used as scalar underneath, where they do scalar-ish things like serving as for-loop constraints, getting passed into `range`, and etc.  So, a significant rewrite would be required if I wanted to let `rows_in` to be a tensor.

**Review by benoitsteiner — APPROVED** [review]

**Comment by benoitsteiner** [comment]
Jenkins, test this please.

**Pull request merged**